packages/functions/src/donor/CancelAppointmentHandler.ts

```diff
@@ -1,41 +1,9 @@
-import {
-  getAppointmentsByIds,
-  setAppointment,
-} from "../dal/AppointmentDataAccessLayer";
 import { FunctionsApi } from "@zm-blood-components/common";
-import { getDonor } from "../dal/DonorDataAccessLayer";
-import { notifyOnCancelAppointment } from "../notifications/CancelAppointmentNotifier";
-import * as DbAppointmentUtils from "../utils/DbAppointmentUtils";
+import { cancelAppointment } from "../common/CancelAppointmentHelper";
 
 export default async function (
   request: FunctionsApi.CancelAppointmentRequest,
   callerId: string
 ) {
-  const donorId = callerId;
-
-  if (!request.appointmentId) {
-    throw new Error("No appointment to cancel");
-  }
-
-  const appointmentToCancel = await getAppointmentsByIds([
-    request.appointmentId,
-  ]);
-  if (appointmentToCancel.length !== 1) {
-    throw new Error("Appointment not found");
-  }
-
-  const appointment = appointmentToCancel[0];
-  if (appointment.donorId !== donorId) {
-    throw new Error("Appointment to be deleted is not booked by donor");
-  }
-
-  const donor = await getDonor(donorId);
-  notifyOnCancelAppointment(appointment, donor!).catch((e) =>
-    console.error("Error notifying on cancelled appointment", appointment.id, e)
-  );
-
-  const updatedAppointment =
-    DbAppointmentUtils.removeDonorFromDbAppointment(appointment);
-
-  await setAppointment(updatedAppointment);
+  await cancelAppointment(callerId, request, true);
```

Comment:
Instead of extracting this to a file and call it with true or false for notifying, I suggest something else:

1. Fetch appointment
2. Extract validation logic to util (like you already did, with small changes)
3. Remove donor and save appointment.
4. Notify user.

Then, in the other function, just run steps 1-3

Reply:
Seems undry